packages/mui-base/src/useMenu/menuItem.types.ts

```diff
@@ -0,0 +1,11 @@
+export interface MenuItemMetadata {
+  id: string;
+  disabled: boolean;
+  label?: string;
+  ref: React.RefObject<HTMLElement>;
+}
+
+export interface MenuItemState {
+  disabled: boolean;
+  highlighted: boolean;
+}
```

Comment:
Call-out that I had to refactor these types from `useMenu.types.ts` as a circular dependency error was occurring between the newly added `UseMenuReturnValue` type within `useMenu.type.ts` as that interface imported from `MenuUnstyledContext.ts` which in turn imported the above two types from `useMenu.type.ts`.

Reply:
Let's move these types into `src/MenuItemUnstyled/MenuItemUnstyled.types.ts`

Reply:
Thanks for the review. Have amended PR